db/product_test.go

```diff
@@ -0,0 +1,135 @@
+package db
+
+import (
+	"context"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/jmoiron/sqlx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type ProductsTestSuite struct {
+	suite.Suite
+	dbStore Storer
+	db      *sqlx.DB
+	sqlmock sqlmock.Sqlmock
+}
+
+func (suite *ProductsTestSuite) SetupTest() {
+	dbStore, dbConn, sqlmock := InitMockDB()
+	suite.dbStore = dbStore
+	suite.db = dbConn
+	suite.sqlmock = sqlmock
+}
+
+func (suite *ProductsTestSuite) TearDownTest() {
+	suite.db.Close()
+}
+
+func (suite *ProductsTestSuite) TestCreateNewProductSuccess() {
+	product := Product{
+		Name:         "test user",
+		Description:  "test database",
+		Price:        123,
+		Discount:     10,
+		Tax:          0.5,
+		Quantity:     5.0,
+		CategoryId:   1,
+		CategoryName: "testing",
+		Brand:        "testing",
+		Color:        "test",
+		Size:         "heigh",
+		URLs:         []string{"url1", "url2"},
+	}
+
+	suite.sqlmock.ExpectBegin()
+
+	suite.sqlmock.ExpectExec("INSERT INTO product").
```

Comment:
This is just suggestion. In project we should avoid using mocks for db methods
You guys can keep this comment as last task if time permits u can just remove mock for db